lib/util/CompatibilityUtil.js

```diff
@@ -0,0 +1,46 @@
+import { isFunction } from 'min-dash';
+
+// Wraps APIs to check:
+// 1) If a callback is passed -> Warn users about callback deprecation.
+// 2) If Promise class is implemented in current environment.
+export function wrapForCompatibility(api) {
+
+  return function() {
+
+    if (!window.Promise) {
+      throw new Error('Promises is not supported in this environment. Polyfill Promises.');
+    }
+
+    var argLen = arguments.length;
+    if (argLen >= 1 && isFunction(arguments[argLen - 1])) {
+
+      var callback = arguments[argLen - 1];
+
+      console.warn(
+        'Passing callbacks to ' + api.name + ' is deprecated and will be removed in a future major release.' +
+        'Please switch to promises, cf. https://bpmn.io/l/moving-to-promises.html'
+      );
+
+      // We want to call the API without the callback.
+      // Arguments is not an Array, but an array like object. That's why
+      // methods like pop() and slice() do not exist within arguments.
+      delete arguments[argLen - 1];
+
+      api.apply(this, arguments).then(function(result) {
+
+        var firstKey = Object.keys(result)[0];
+
+        // The APIs we are wrapping all resolve a single item depending on the API.
+        // For instance, importXML resolves { warnings } and saveXML returns { xml }.
+        // That's why we can call the callback with the first item of result.
```

Comment:
Why do we actually wrap the result in an object? Maybe I missed sth...

Reply:
Shouldn't `importXML` resolve with `err` and `definitions`?

Reply:
Currently, `done` is called with `err, warnings`
https://github.com/bpmn-io/bpmn-js/blob/develop/lib/BaseViewer.js#L89

Reply:
For definitions we have importDefinitions.

We wrap the result in an object as a design choice. We figured it'd be easier like that to add stuff later.

Reply:
Ah, I forgot.

Reply:
OK